vm-setup/roles/v1aX_integration_test/filter_plugins/k8s_fields.py

```diff
@@ -0,0 +1,36 @@
+"""
+Filter out managed fields, uids, timestamps and similar from Kubernetes
+resources. These fields are often "in the way" when trying to find the relevant
+parts or when "restoring" resources to a new cluster.
+"""
+
+import copy
+
+
+def k8s_backup(resources):
+    """
+    Take a backup of k8s resources by stripping away problematic fields.
+    Fields that are automatically created and should normally not be touched
+    will be removed so that the resources can safely be restored from this
+    "backup".
+    Removed fields under .metadata: uid, managedFields, resourceVersion,
+    creationTimestamp
+    """
```

Comment:
k8s `apply` https://docs.ansible.com/ansible/latest/collections/community/kubernetes/k8s_module.html#parameter-apply supposed to ignore properties that are automatically generated. Does not it help with those fields ?

Reply:
Interesting! Will check, thanks!

Reply:
Unfortunately it doesn't work. :slightly_frowning_face: I guess it ignores the auto generated properties when doing the comparison, but not when doing the actual update. This is how I tested it:

```yaml
- name: Create deployment
      kubernetes.core.k8s:
        api_version: apps/v1
        kind: Deployment
        name: test
        namespace: default
        resource_definition:
          metadata:
            labels:
              app: test
          spec:
            replicas: 1
            selector:
              matchLabels:
                app: test
            template:
              metadata:
                labels:
                  app: test
              spec:
                containers:
                  - name: nginx
                    image: nginx

    - name: Get deployment
      kubernetes.core.k8s_info:
        api_version: apps/v1
        kind: Deployment
        name: test
        namespace: default
      register: deployment

    - debug:
        msg: "{{ deployment }}"

    - name: Scale deployment
      kubernetes.core.k8s_scale:
        api_version: apps/v1
        kind: Deployment
        name: test
        namespace: default
        replicas: 2

    - name: Restore deployment
      kubernetes.core.k8s:
        api_version: apps/v1
        kind: Deployment
        name: test
        namespace: default
        apply: yes
        resource_definition: "{{ deployment.resources[0] }}"
```

In the last step there is always a failure due to status 409 (conflict). I believe this is because the resource version has changed. So the server knows that we are trying to update an object that we haven't seen the latest version of yet.